File: LeetCode/DataStruct/Golang/Leetcode2104-子数组范围和.go
package sbgo

func subArrayRanges(nums []int) int64 {
	res, n := int64(0), len(nums)
	for i := 0; i < n; i++ {
		mx, mn := nums[i], nums[i]
		for j := i + 1; j < n; j++ {
			if nums[j] > mx {
				mx = nums[j]
			}
			if nums[j] < mn {
				mn = nums[j]
			}

			res += (int64)(mx - mn)
		}
	}
	return res
}

File: LeetCode/DataStruct/Python/Leetcode2104-子数组范围和.py
class Solution:
    def subArrayRanges(self, nums: List[int]) -> int:
        res, n = 0, len(nums)
        for i in range(n):
            mx, mn = nums[i], nums[i]
            for j in range(i + 1, n):
                mx = max(mx, nums[j])
                mn = min(mn, nums[j])
                res += mx - mn
        return res
